src/main/java/frc/robot/subsystems/TrapArmSubsystem.java

```diff
@@ -0,0 +1,224 @@
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.TrapArmConstants;
+
+public class TrapArmSubsystem extends SubsystemBase {
+  // Wrist motors
+  private final CANSparkMax wristMotor;
+  private final CANSparkMax rollerMotor;
+
+  // Elevator motors
+  private final CANSparkMax baseMotor1;
+  private final CANSparkMax baseMotor2;
+  private final CANSparkMax scoringMotor;
+
+  // Beam Breaks
+  private final DigitalInput sourceBreak;
+  private final DigitalInput groundBreak;
+
+  // Limit Switches
+  private final DigitalInput baseBreak;
+  private final DigitalInput scoringBreak;
+
+  // Encoders
+  private final CANcoder wristEncoder;
+  // PID
+  // P, I, D, Iz, FF
+  private Double[] basePID = {36e-5, 5e-7, 1e-4, 0.0, 2e-6};
+  private Double[] scoringPID = {36e-5, 5e-7, 1e-4, 0.0, 2e-6};
+
+  // States
+  public static enum TrapArmState {
+    STOWED(0.0, 0.0, 0.0),
+    FROM_INTAKE(0.3, 0.0, 0.0),
+    FROM_SOURCE(0.4, 0.0, 1.0),
+    TRAP_SCORE(0.0, 1.0, 2.0),
+    AMP_SCORE(0.0, 0.0, 0.0);
+
+    private Double wristPose;
+    private Double basePose;
+    private Double scoringPose;
+
+    TrapArmState(Double wrist, Double base, Double scoring) {
+      this.wristPose = wrist;
+      this.basePose = base;
+      this.scoringPose = scoring;
+    }
+
+    public Double getWristPose() {
+      return wristPose;
+    }
+
+    public Double getBasePose() {
+      return basePose;
+    }
+
+    public Double getScoringPose() {
+      return scoringPose;
+    }
+  }
+
+  /** Creates a new TrapArm. */
+  public TrapArmSubsystem() {
+    // Wrist
+    wristMotor = new CANSparkMax(TrapArmConstants.wristMotor_ID, MotorType.kBrushless);
+    wristMotor.restoreFactoryDefaults();
+    wristEncoder = new CANcoder(6);
+
+    rollerMotor = new CANSparkMax(TrapArmConstants.rollerMoter_ID, MotorType.kBrushless);
+    rollerMotor.restoreFactoryDefaults();
+
+    sourceBreak = new DigitalInput(TrapArmConstants.sourceBreak_ID);
+    groundBreak = new DigitalInput(TrapArmConstants.groundBreak_ID);
+
+    // Arm
+    baseMotor1 = new CANSparkMax(TrapArmConstants.baseMotor1_ID, MotorType.kBrushless);
+    baseMotor1.restoreFactoryDefaults();
+    baseMotor1.getPIDController().setP(basePID[0]);
+    baseMotor1.getPIDController().setI(basePID[1]);
+    baseMotor1.getPIDController().setD(basePID[2]);
+    baseMotor1.getPIDController().setIZone(basePID[3]);
+    baseMotor1.getPIDController().setFF(basePID[4]);
+
+    baseMotor2 = new CANSparkMax(TrapArmConstants.baseMotor2_ID, MotorType.kBrushless);
+    baseMotor2.restoreFactoryDefaults();
+    baseMotor2.getPIDController().setP(basePID[0]);
+    baseMotor2.getPIDController().setI(basePID[1]);
+    baseMotor2.getPIDController().setD(basePID[2]);
+    baseMotor2.getPIDController().setIZone(basePID[3]);
+    baseMotor2.getPIDController().setFF(basePID[4]);
```

Comment:
I feel like basemotor will follow each other. But not sure

Reply:
I had problems when prototyping when just using follow.